mars/services/storage/transfer.py

```diff
@@ -0,0 +1,116 @@
+    data: bytes
+    session_id: str
+    data_key: str
+    level: StorageLevel
+    is_eof: bool
+
+
+class SenderManagerActor(mo.Actor):
+    async def __post_create__(self):
+        self._storage_manager_ref: Union[mo.ActorRef, StorageManagerActor] = \
+            await mo.actor_ref(self.address, StorageManagerActor.default_uid())
+        self._storage_handler: Union[mo.ActorRef, StorageHandlerActor] = \
+            await mo.actor_ref(self.address, StorageHandlerActor.default_uid())
+
+    @staticmethod
+    async def get_receiver_ref(address: str):
+        return await mo.actor_ref(
+            address=address, uid=ReceiverManagerActor.default_uid())
+
+    @extensible
+    async def send_data(self,
+                        session_id: str,
+                        data_key: str,
+                        address: str,
+                        level: StorageLevel,
+                        block_size: int = None):
+        block_size = block_size or options.worker.transfer_block_size
+        receiver_ref = await self.get_receiver_ref(address)
+        info = await self._storage_manager_ref.get_data_info(session_id, data_key)
+        store_size = info.store_size
+        await receiver_ref.open_writer(session_id, data_key,
+                                       store_size, level)
+        async with await self._storage_handler.open_reader(
+                session_id, data_key) as reader:
+            while True:
+                part_data = await reader.read(block_size)
+                is_eof = len(part_data) < block_size
+                message = TransferMessage(part_data, session_id, data_key, level, is_eof)
+                send_task = asyncio.create_task(receiver_ref.receive_part_data(message))
+                yield send_task
+                if is_eof:
+                    break
+
+
+class ReceiverManagerActor(mo.Actor):
+    def __init__(self):
+        self._key_to_writer_info = dict()
+
+    async def __post_create__(self):
+        self._storage_manager_ref: Union[mo.ActorRef, StorageManagerActor] = \
+            await mo.actor_ref(self.address, StorageManagerActor.default_uid())
+        self._storage_handler: Union[mo.ActorRef, StorageHandlerActor] = \
+            await mo.actor_ref(self.address, StorageHandlerActor.default_uid())
+
+    async def open_writer(self,
+                          session_id: str,
+                          data_key: str,
+                          data_size: int,
+                          level: StorageLevel):
+        try:
+            await self._storage_manager_ref.allocate_size(data_size, level)
+            writer = await self._storage_handler.open_writer(session_id, data_key,
+                                                             data_size, level)
+            self._key_to_writer_info[(session_id, data_key)] = (writer, data_size, level)
+        except asyncio.CancelledError:  # pragma: no cover
+            await self._storage_manager_ref.release_quota(
```

Comment:
Do we need a unified cleanup method to handle both successes and errors, for instance writer cleanup? something is surely missing here.